repo_link/repo_link.py

```diff
@@ -28,23 +28,34 @@ def clone_link(self):
 def parse(link: str) -> RepoData:
     """Given a github link returns the File path, git commit, line and collumn the link refers to"""
     # https://regex101.com/r/jBJ7PI/4
-    link_pattern = r"(?P<base_link>https://github\.com)/(?P<user>.+)/(?P<repository>[^/]+)/blob/(?P<commit>[^/]+)/(?P<path>[^#]+)(?:#L(?P<line>\d+))?"
+    link_pattern = r"(?P<base_link>https://github\.com)/(?P<user>.+)/(?P<repository>[^/]+)/blob/(?P<branch_and_path>[^#]+)(?:#L(?P<line>\d+))?"
     match = re.match(link_pattern, link)
     if match:
         groups = match.groups()
         return RepoData(**match.groupdict())
     raise ValueError("Invalid link")
 
 
-def checkout(repo: Repo, commit: str):
+def checkout(repo: Repo, branch: str):
```

Comment:
Split this into finding the path/commit, and checkout afterwards

Reply:
Didn't get a chance yet to try this. Any idea why `commit` with the correct branch name throws an error?